shipper/shipper.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+option go_package = "github.com/elastic/elastic-agent-shipper/shipper";
```

Comment:
I'd prefer if this lives in an `api` or a `proto` directory than one just called shipper, since it has vendored code in it.

Reply:
Yah, agreed, wasn't sure what to call it at the time.